extensions/cornerstone/src/registerCustomCornerstoneTools.ts

```diff
@@ -0,0 +1,132 @@
+import { toolNames } from './initCornerstoneTools';
+import { registerCustomToolsMappingFactory } from './utils/measurementServiceMappings/customToolsMappingFactory';
+import * as csTools from '@cornerstonejs/tools';
+
+const { BaseTool } = csTools;
+
+/**
+ * Object to store registered Custom Tools (classes)
+ */
+const customTools = {};
+
+/**
+ * Auxiliary class to store original customTools/toolNames to be restore on a fallback approach.
+ */
+class CustomToolCache {
+  constructor() {
+    this.customTools = {};
+    this.toolNames = {};
+  }
+
+  setTo(toolName, from, to) {
+    if (from[toolName]) {
+      to[toolName] = from[toolName];
+    }
+  }
+
+  addToCache(CustomTool) {
```

Comment:
this can be just `add` since the className is `customToolCache` so it will be `customToolCache.add` instead of `customToolCache.addToCache` same for revert